plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -110,14 +114,14 @@ public static List<DeltaLakeColumnHandle> extractPartitionColumns(MetadataEntry
         return extractPartitionColumns(extractSchema(metadataEntry, typeManager), metadataEntry.getCanonicalPartitionColumns());
     }
 
-    public static List<DeltaLakeColumnHandle> extractPartitionColumns(List<ColumnMetadata> schema, List<String> canonicalPartitionColumns)
+    public static List<DeltaLakeColumnHandle> extractPartitionColumns(List<DeltaLakeColumnMetadata> schema, List<String> canonicalPartitionColumns)
     {
         if (canonicalPartitionColumns.isEmpty()) {
             return ImmutableList.of();
         }
         return schema.stream()
                 .filter(entry -> canonicalPartitionColumns.contains(entry.getName()))
-                .map(entry -> new DeltaLakeColumnHandle(entry.getName(), entry.getType(), PARTITION_KEY))
+                .map(entry -> new DeltaLakeColumnHandle(entry.getName(), entry.getType(), entry.getPhysicalName(), entry.getPhysicalColumnType(), PARTITION_KEY))
```

Comment:
Yes, it can. Here is sample metadata json generated by `testShowStatsOnPartitionedForColumnMappingModeName`. `schemaString` has physicalName for a partition column.
```json
{
    "metaData": {
        "id": "209353e0-0bbe-4511-ba61-4528a304cd69",
        "format": {
            "provider": "parquet",
            "options": []
        },
        "schemaString": "{\"type\":\"struct\",\"fields\":[{\"name\":\"a_number\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{\"delta.columnMapping.id\":1,\"delta.columnMapping.physicalName\":\"col-3a72ac0b-27cf-48af-86e4-66ffcc3b4773\"}},{\"name\":\"part\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{\"delta.columnMapping.id\":2,\"delta.columnMapping.physicalName\":\"col-b8ecccee-2335-4c55-b1fe-a7b0b0fcd477\"}}]}",
        "partitionColumns": [
            "part"
        ],
        "configuration": {
            "delta.columnMapping.mode": "name",
            "delta.columnMapping.maxColumnId": "2"
        },
        "createdTime": 1656041255998
    }
}
```